File: Frontend/src/lib/services/pairs.ts
import { Pair } from "@/interfaces/entities/pair";
import { Result } from "@/types/result";
import { invoke } from "@tauri-apps/api/core";

export async function select_pairs(env: string): Promise<Result<Pair[]>> {
    try {
        let data = await invoke<Pair[]>("select_pairs", {
            env,
        });

        return { ok: true, data };
    } catch (error) {
        return { ok: false, error };
    }
}

File: Frontend/src-tauri/src/command/pairs.rs
use crate::{
    core::select_pairs_core,
    models::{entities::pairs::Model, structs::PairRequest},
};

#[tauri::command]
pub async fn select_pairs(env: String, pair: Option<PairRequest>) -> Result<Vec<Model>, String> {
    select_pairs_core(env, pair).await
}
